rust-runtime/aws-smithy-http-client/src/client/tls.rs

```diff
@@ -15,9 +15,9 @@ pub enum Provider {
     ))]
     /// TLS provider based on [rustls](https://github.com/rustls/rustls)
     Rustls(rustls_provider::CryptoMode),
-    // TLS provider based on [S2N](https://github.com/aws/s2n-tls)
-    // S2n,
-    // TODO(hyper1) s2n support
+    /// TLS provider based on [s2n-tls](https://github.com/aws/s2n-tls)
+    #[cfg(feature = "s2n-tls")]
```

Comment:
Do we need to worry about or consider that s2n only works on unix platforms? Should this cfg be gated on both the feature and platform support?

Reply:
Yeah I considered doing that as well. My concern was someone enabling s2n-tls on a non-unix platform and then trying to figure out why they can't select the s2n-tls provider. A build failure after enabling the feature seemed like it might be less confusing. But maybe this would also be confusing. Let me know what you think, I'm happy to change it!

Reply:
We can leave it for now unless someone else feels strongly at the moment.